kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -296,7 +305,7 @@ protected TransactionImpl buildTransactionInternal(
 
     // We use the existing clustering columns to validate schema evolution
     Optional<List<Column>> existingClusteringCols =
```

Comment:
@KaiqiJinWow can you help explain this a bit for me?

so we have `initialClusteringColumns` and `existingClusteringColumns`

I think `initialClusteringColumns` really means `newClusteringColumnsInThisTxn`

but why is it that: if this is an UPDATE (not create, not replace) and `newClusteringColumnsInThisTxn`, then we set `existingClusteringCols` to EMPTY?

wouldn't the `existingClusteringCOls` during an UPDATE be `ClusteringUtils.getClusteringColumnsOptional(latestSnapshot.get());`